### ISSUE TITLE
Do not wait for enter or ctrl+c to stop

### ISSUE THREAD

**Issue by Osiris-Team** [mvysny/vaadin-boot#26]
I have my own console input reader which handles the application state, thus `System.in.read() == -1` from VaadinBoot kind of breaks this.
I tried to override the run method to remove it, but the private fields also prevent that, maybe having them protected would work?
Or is there another workaround for this? (For now I simply removed the code that depends on private fields)
Also having `Env` class public would be great too.

![image](https://github.com/user-attachments/assets/08795cf5-f9e6-4072-bc60-0d97e81dec18)

**Comment by mvysny** [comment]
Hi, would calling `start()`/`stop()` help?

```java
VaadinBoot boot = new VaadinBoot();
boot.start();
// wait for the Enter key
boot.stop();
```

**Comment by Osiris-Team** [comment]
@mvysny thats basically what I did, however having access to Env and those private fields related to dev mode would be nice.

**Comment by mvysny** [comment]
I've added public getters for those private fields in the newest master (not yet released), so this topic should be sorted out. Please check out the newest sources and let me know whether this solves the issue for you.

 Also, it sounds like the `start()`/`stop()` function pair solves the original issue, so I'm poised to close this as "works as intended".

The `Env` class is kind of a helper class, not intended to be a part of the public API, that's why it's not made public. Yet, I'm of course willing to make it public if it can be shown that it contains functionality that's critical to solve this issue.

@Osiris-Team does the `start()`/`stop()` fix the issue completely for you, or is there something I need to do? If yes, could you provide a concrete example of what needs to be done?

**Comment by Osiris-Team** [comment]
Great, thanks!

**Comment by mvysny** [comment]
Fixed and released as 12.3
